test/dataframerow.jl

```diff
@@ -586,4 +586,68 @@ end
     @test_throws ArgumentError r .+ 1
 end
 
+@testset "comparison tests: DataFrameRow, NamedTuple and GroupKey" begin
+    df = DataFrame(a=[1, 2], b=[missing, 3])
+    dfr = [df[1, :], df[2, :], df[1, 1:1], df[2, 1:1]]
+    nt = NamedTuple.(dfr)
+    gk = [keys(groupby(df, [:a, :b])); keys(groupby(df, :a))]
+
+    for l in (dfr[1], nt[1], gk[1]), r in (dfr[1], nt[1], gk[1])
+        @test ismissing(l == r)
+        @test ismissing(l == (a=1, b=2))
+        @test l ≅ r
+        @test l ≇ (a=1, b=2)
+        # work around https://github.com/JuliaLang/julia/pull/40147
+        if !(l isa NamedTuple && r isa NamedTuple)
+            @test ismissing(l < r)
```

Comment:
It could be interesting to test the equivalent of `(1, missing) < (2, 1)`, which is a corner case that shouldn't give `missing` but is easy to get wrong.

Reply:
I will add it